### PR TITLE
Remove spaces from Merkle proof data structure names

### PR DESCRIPTION
**Pull request by adlerjohn**
- [x] Remove spaces for proof names, making them consistent with other structs
- [x] Make signed transaction data be suffixed with type

### CODE REVIEW

**Comment by adlerjohn** [comment]
> [Have you also updated the names in the references to these data structures in the tables?](https://github.com/lazyledger/lazyledger-specs/pull/84#issuecomment-707962891)

The Merkle proofs whose name were changed aren't used anywhere yet, so there was nothing to update.

**Comment by adlerjohn** [comment]
FYI, it appears that the GitHub action is getting cancelled for whatever reason. You can run the link checker locally: https://github.com/tcort/markdown-link-check, which should show that there are no broken links.

**Comment by musalbas** [comment]
How about https://github.com/lazyledger/lazyledger-specs/blob/a87e5b54d3ac4a8940575dd866e656212985685d/specs/data_structures.md#invalid-erasure-coding?

**Comment by adlerjohn** [comment]
> [How about...](https://github.com/lazyledger/lazyledger-specs/pull/84#issuecomment-708020318)

Missed those, good catch. Fixed in e71c918.

File: specs/data_structures.md [hunk 1]
```diff
@@ -424,22 +426,22 @@ If a malicious block producer incorrectly computes the 2D Reed-Solomon code for
 
 #### ShareProof
 
-| name       | type                                                         | description                                                                                       |
-| ---------- | ------------------------------------------------------------ | ------------------------------------------------------------------------------------------------- |
-| `share`    | [Share](#share)                                              | The share.                                                                                        |
-| `proof`    | [Namespace Merkle Tree Proof](#namespace-merkle-tree-proofs) | The Merkle proof of the share in [`availableDataRoot`](#header).                                  |
```
Comment (musalbas) [inline specs/data_structures.md:430]: Shouldn't the name be changed too?

File: specs/data_structures.md [hunk 2]
```diff
@@ -424,22 +426,22 @@ If a malicious block producer incorrectly computes the 2D Reed-Solomon code for
...
-| `isCol`       | `bool`                                                       | A Boolean indicating if it is an offending row or column; `false` if it is a row. |
-| `position`    | `uint64`                                                     | The index of the row or column in the square.                                     |
+| name          | type                                                     | description                                                                       |
+| ------------- | -------------------------------------------------------- | --------------------------------------------------------------------------------- |
+| `shareProofs` | [ShareProof](#shareproof)`[]`                            | The available shares in the offending row or column.                              |
+| `root`        | [HashDigest](#hashdigest)                                | The Merkle root of the offending row or column.                                   |
+| `proof`       | [Namespace Merkle Tree Proof](#namespacemerkletreeproof) | The Merkle proof of the row or column root in [`availableDataRoot`](#header).     |
```
Comment (musalbas) [inline specs/data_structures.md:442]: Shouldn't the names be changed too?

Comment (adlerjohn) [reply specs/data_structures.md]: Fixed in 8cdb742.

**Review by musalbas — APPROVED** [review]

**Pull request merged**